modules/apps/frontend-js/frontend-js-bundle-config-extender/src/main/java/com/liferay/frontend/js/bundle/config/extender/internal/servlet/taglib/JSBundleConfigTopHeadDynamicInclude.java

```diff
@@ -128,6 +136,34 @@ protected void setJSBundleConfigTracker(
 		_jsBundleConfigTracker = jsBundleConfigTracker;
 	}
 
+	private String _getModuleMain(JSBundleConfigTracker.JSConfig jsConfig) {
+		ServletContext servletContext = jsConfig.getServletContext();
+
+		try {
+			URL url = servletContext.getResource("package.json");
+
+			if (url != null) {
+				try (InputStream inputStream = url.openStream()) {
+					JSONObject jsonObject = _jsonFactory.createJSONObject(
+						StringUtil.read(inputStream));
+
+					String moduleName = jsonObject.getString("name");
+					String moduleVersion = jsonObject.getString("version");
+					String moduleMainFile = ModuleNameUtil.toModuleName(
```

Comment:
Should we check if `main` is missing and use `/index.js` in that case? Otherwise there's the risk of throwing a NPE, I think...

Reply:
Thanks for the advice @izaera, sounds like a good idea.
Would something like this be OK? :see_no_evil: 

```suggestion
					String moduleMainFile = jsonObject.getString("main");
					// Didn't show it here, but I did import `com.liferay.portal.kernel.util.Validator;`
					if (Validator.isNull(moduleMainFile)) {
						moduleMainFile = "index.js";
					}

					return StringBundler.concat(
						moduleName, "@", moduleVersion, "/", 
						ModuleNameUtil.toModuleName(moduleMainFile));
```

Reply:
That said, isn't it better for `main` to be missing if it really doesn't exist? 

I did test in modules with a `package.json` that don't specify `main` (for example: [this](https://github.com/liferay/liferay-portal/blob/ce10eb3c2e2f3427b904c078ac8192ca5b7e8d37/modules/apps/portal-workflow/portal-workflow-task-web/package.json) one) and didn't see an NPE. What "bothers" me is that by saying that `main` is `index.js` when it actually isn't is that we might be confusing the user because we're making this up. Sorry if I'm understanding this incorrectly, I get that there's not easy way to solve this.

Reply:
In Java we're loading the `package.json` file, after checking if exists. We're not loading the main file here so no NPE should happen. What could happen is a JS error if you configure a module's `config.file` without having a `main` in the `pagacke.json`, but I think that's a configuration problem and it's fine that fails in that case. What do u think?

Reply:
That's more a less the way I understood it as well, so I'm OK with that explanation, let's see if Iván has more things to say.

Reply:
`main` is `index.js` by default in Node, so a `package.json` without `main` should behave like a `package.json` with `main: "index.js"`

Reply:
So, I would commit the proposed change @julien ...